pkg/generator/generator.go

```diff
@@ -273,17 +275,28 @@ func (c *GenVars) keySeparatorLookup(key, val string) string {
 		return val
 	}
 
-	pm := ParsedMap{}
+	keys, err := ajson.JSONPath([]byte(val), "$.."+kl[1])
+	if err != nil {
+		panic(err)
+	}
 
-	if ok := isParsed(val, &pm); ok {
-		log.Debugf("attempting to find by key: %v in value: %v", kl, val)
-		if foundVal, ok := pm[kl[1]]; ok {
-			log.Debugf("found by key: %v, in value: %v, of: %v", kl[1], val, foundVal)
-			return fmt.Sprintf("%v", foundVal)
+	for _, v := range keys {
+		switch v.Type() {
```

Comment:
I think we should assume that there will only ever be a single match. and can probably avoid init a for loop.